### PR TITLE
Problem: no efficient bank api to support evm add/sub balances

### PR DESCRIPTION
**Pull request by yihuang**
# Description

Add `AddBalance`/`SubBalance` low level api to bank keeper, to support evm integration more efficiently.

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 * (baseapp) [#205](https://github.com/crypto-org-chain/cosmos-sdk/pull/205) Add `TxExecutor` baseapp option, add `TxIndex`/`TxCount`/`MsgIndex`/`BlockGasUsed` fields to `Context, to support tx parallel execution.
 * (baseapp) [#205](https://github.com/crypto-org-chain/cosmos-sdk/pull/205) Support mount object store in baseapp, add `ObjectStore` api in context..
+* (x/bank) [#]() Add low level `AddBalance`,`SubBalance` APIs to bank keeper.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (x/bank) [#239](https://github.com/crypto-org-chain/cosmos-sdk/pull/239) Add low level `AddBalance`,`SubBalance` APIs to bank keeper.
```

File: x/bank/keeper/keeper.go [hunk 2]
```diff
@@ -419,6 +419,33 @@ func (k BaseKeeper) BurnCoins(ctx context.Context, moduleName string, amounts sd
...
+// caller should make sure the total supply of the denom not changed and the account exists.
+// it's for evm integration, call at your own risk.
+// it emits mint event.
+func (k BaseKeeper) AddBalance(ctx context.Context, addr sdk.AccAddress, coin sdk.Coin) error {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// emit mint event
+	sdkCtx.EventManager().EmitEvent(
+		types.NewCoinMintEvent(addr, sdk.NewCoins(coin)),
+	)
+	return k.addCoin(ctx, addr, coin)
```
Comment (mmsqe) [inline x/bank/keeper/keeper.go]: might handle add sub coin first, we don't need emit event if error right?

Comment (yihuang) [reply x/bank/keeper/keeper.go]: it's always executed in a cache context, it's atomic anyway.

Comment (yihuang) [reply x/bank/keeper/keeper.go]: reordered, it make sense in the sense of efficiency

**Review by mmsqe — APPROVED** [review]

**Pull request merged**